testing/litetest/lib/src/test_suite.dart

```diff
@@ -0,0 +1,133 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.12
+
+import 'dart:async';
+import 'dart:collection';
+import 'dart:io' show exit, stdout;
+import 'dart:isolate';
+
+import 'package:async_helper/async_minitest.dart' as m;
+
+import 'test.dart';
+
+/// A suite of tests, added with the [group] and [test] methods, which will be
+/// run in a following event.
+class TestSuite {
+  /// Creates a new [TestSuite] with logs written to [logger] and callbacks
+  /// given by [lifecycle].
+  TestSuite({
+    StringSink? logger,
+    Lifecycle? lifecycle,
+  }) :
+    _logger = logger ?? stdout,
+    _lifecycle = lifecycle ?? _DefaultLifecycle();
+
+
+  final Lifecycle _lifecycle;
+  final StringSink _logger;
+  final Queue<String> _groupStack = Queue<String>();
+  bool _testQueuePrimed = false;
+  final Queue<Test> _testQueue = Queue<Test>();
+  final Map<String, Test> _runningTests = <String, Test>{};
+
+  /// Adds a test to the test suite.
+  void test(
+    String name,
+    FutureOr<void> Function() body, {
```

Comment:
`FutureOr<void>` is a little dubious (https://github.com/dart-lang/sdk/issues/35237#issuecomment-440660116). But package:test uses `dynamic Function() body`, which I'm not sure is much better. However, it seems like if we were to run analysis on this, it would end up complaining about a missing return based on that issue, and that this is by design. 

We should probably either use `dynamic Function() body`.

Reply:
Went with `dynamic Function()`.